transformer_engine/pytorch/module/linear.py

```diff
@@ -574,6 +591,8 @@ class Linear(TransformerEngineBaseModule):
                              have an additional `main_grad` attribute (used instead of the
                              regular `grad`) which is a pre-allocated buffer of the correct
                              size to accumulate gradients in.
+    cpu_offloading : bool, default = 'False'
+                    if set to `True`, offloads the input activation to this module to the CPU.
```

Comment:
This is not consistent with the actual parameters passed to __init__.

What I would do here is something like this:
in `cpu_offloading.py`:
```python
CPUOffloadEnabled = False

class CPUOffloadSaveTensorHook:
    def __enter__():
        global CPUOffloadEnabled
        CPUOffloadEnabled = True
        # the rest of the logic
    def __exit__():
        global CPUOffloadEnabled
        CPUOffloadEnabled = False
        # the rest of the logic
```
then in each module you would not have any parameter, but it would just check the state during `__init__`:
```python
self.cpu_offloading = CPUOffloadEnabled
```

Reply:
We discussed to pass ContextManager to __init__ of the modules earlier right, am I missing something? 

I can change the description to match the ContextManager expectations.